pkg/process/metadata/parser/nodejs/nodejs.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package nodejs wraps functions to guess service name for node applications
+package nodejs
+
+import (
+	"encoding/json"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+type packageName struct {
+	// Name of the package in the json file
+	Name string `json:"name,omitempty"`
+}
+
+// FindNameFromNearestPackageJSON finds the package.json walking up from the absFilePath.
+// if a package.json is found, returns the value of the field name if declared
+func FindNameFromNearestPackageJSON(absFilePath string) (bool, string) {
+	current := filepath.Dir(absFilePath)
+	up := filepath.Dir(current)
+	for run := true; run; run = current != up {
+		ok, value := maybeExtractServiceName(filepath.Join(current, "package.json"))
+		if ok {
+			return ok && len(value) > 0, value
```

Comment:
I think the result of `maybeExtractServiceName` should be solid, meaning if moving `ok && len(value) > 0` inside `maybeExtractServiceName`

Reply:
I understand the suggestion however the boolean returned in the two function has not the same meaning:
* `maybeExtractServiceName` return a bool indicating whenever or not the recursion should stop (e.g. I found that file)
* `FindNameFromNearestPackageJSON` return true if a valid value has been extracted